libs/libarchfpga/src/physical_types.h

```diff
@@ -179,9 +179,18 @@ enum e_pb_type_class {
     UNKNOWN_CLASS = 0,
     LUT_CLASS = 1,
     LATCH_CLASS = 2,
-    MEMORY_CLASS = 3
+    MEMORY_CLASS = 3,
+    NUM_CLASSES
 };
 
+// Set of all pb_type classes
+constexpr std::array<e_pb_type_class, NUM_CLASSES> TYPE_CLASSES = {
```

Comment:
We have a lot of different kinds of classes (pin classes, etc.).  So I think we should name this array PB_TYPE_CLASSES to be a bit more specific, and similarly name the char* array PB_TYPE_CLASS_STRING.  Other than that the code looks good.

Reply:
Sounds good, done.